plugins/logs/plugin.go

```diff
@@ -767,6 +770,19 @@ func (p *Plugin) loop() {
 func (p *Plugin) doOneShot(ctx context.Context) error {
 	uploaded, err := p.oneShot(ctx)
 
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
```

Comment:
Added this change to address the [race detector](https://github.com/open-policy-agent/opa/actions/runs/4351330130/jobs/7619374942#step:4:222) failure.